src/documents/posts/2015-04-02-delivery-engineering.md

```diff
@@ -0,0 +1,47 @@
+---
+title: Delivery Engineering
+author: Bob Zoller
+layout: post
+disqus:
+  shortname: goodeggsbytes
+  url: "http://bites.goodeggs.com/post/delivery-engineering"
+---
+
+Delivery Engineering is a term I first heard in the January 2015 [Thoughtworks Radar][thoughtworks-deleng]:
+
+> where there *is* a need for significant investment in [DevOps] tooling and automation, we do see a role for a Delivery Engineering team.  Rather than being a helpdesk, these teams build tooling and enable teams to deploy, monitor, and maintain their own production environments.
+
+In other words, Delivery Engineering enables the _culture_ of DevOps to thrive on each team, rather than silo that knowledge up in one team or role.
+
+I was excited when I read this because it concisely captured the goals of a team I had recently formed, which we were calling (quite opaquely), "Infrastructure."  We formed the Infrastructure team because we felt like we had crossed a threshold as an engineering organization, where it made sense to carve out a set of folks that could own shared tools and services used by all our engineering teams.  Think continuous integration pipelines, node modules for reporting stats, best practices for alerting.
+
+So what's working well and what isn't?
+
+<!-- more -->
+
+Having one team who is accountable for shared services has been an obvious win.  This includes both internal shared services (like a custom deploy server we call Ecru), and thirdparty ones too (like [Travis][travis], [MongoLab][mongolab], [SumoLogic][sumologic], etc).  That's not to say we're the gatekeeper -- no, credentials are still shared and teams often reach out for support directly to those companies.  But our team is there to support our peers, manage upgrades, publish best practices, etc.
```

Comment:
Is Ecru open source? We want to do that, right? It seems like it could be valuable for other Travis users. You can add "Ecru - more on that later" and when you have a post or public repo for Ecru add the link in.